cli/azd/pkg/input/console.go

```diff
@@ -84,6 +106,122 @@ func (c *AskerConsole) Message(ctx context.Context, message string) {
 	}
 }
 
+func (c *AskerConsole) MessageUx(ctx context.Context, message string, format MessageUxType) {
+	formattedText, err := addFormat(message, format)
+	// Message and MessageUx don't return errors. Let's log the error and use the original message on error
+	if err != nil {
+		log.Printf("Failed adding format for MessageUx: %s. Using message with no ux format", err.Error())
+		c.Message(ctx, message)
+		return
+	}
+
+	c.Message(ctx, formattedText)
+}
+
+func addFormat(message string, format MessageUxType) (withFormat string, err error) {
+	switch format {
+	case Title:
+		withFormat = output.WithBold(fmt.Sprintf("\n%s\n", message))
+	case ResultSuccess:
+		withFormat = output.WithSuccessFormat("\n%s: %s", "SUCCESS", message)
+	case ResultError:
+		withFormat = output.WithErrorFormat("\n%s: %s", "ERROR", message)
+	default:
+		return withFormat, fmt.Errorf("Unknown UX format type")
+	}
+
+	return withFormat, nil
+}
+
+func (c *AskerConsole) ShowSpinner(ctx context.Context, title string, format MessageUxType) {
+	// make sure spinner exists
+	if c.spinner == nil {
+		c.spinner, _ = yacspin.New(yacspin.Config{
+			Frequency:       200 * time.Millisecond,
+			Writer:          c.writer,
+			Suffix:          " ",
+			SuffixAutoColon: true,
+		})
+	}
+	// If running, pause to apply style changes
+	if c.spinner.Status() == yacspin.SpinnerRunning {
+		_ = c.spinner.Pause()
+	}
+
+	// Update style according to MessageUxType
+	c.spinner.Message(title)
+	_ = c.spinner.CharSet(getCharset(format))
+
+	// unpause if Paused
+	if c.spinner.Status() == yacspin.SpinnerPaused {
+		_ = c.spinner.Unpause()
+	} else if c.spinner.Status() == yacspin.SpinnerStopped {
+		_ = c.spinner.Start()
+	}
+}
+
+func getCharset(format MessageUxType) []string {
+	customCharSet := []string{
+		"|       |", "|=      |", "|==     |", "|===    |", "|====   |", "|=====  |", "|====== |", "|=======|"}
+
+	newCharSet := make([]string, len(customCharSet))
+	for i, value := range customCharSet {
+		newCharSet[i] = fmt.Sprintf("%s%s", getIndent(format), value)
+	}
```

Comment:
It feels like introducing indentation shouldn't be the responsibility of "formatting the progress bar", but rather the spinner would accept a writer that can introduce indentation levels.

Reply:
Unfortunately, once the spinner is created, we can't change the `Writer` .  We need to be able to mutate the indentation and the message from the spinner without re-creating the spinner whenever the indentation value changes.